### ISSUE TITLE
[Question] Difficulty Reproducing Results

### ISSUE THREAD

**Issue by vegetableno1** [microsoft/FlexCAD#2]
Hi authors,

First of all, thank you for your outstanding work and for making the code publicly available. It's a great contribution to the community.

I am currently trying to reproduce the results for the flexcad experiment, but I've encountered a couple of issues and was hoping you might be able to provide some guidance.

# 1. Training Loss Plateaus Early

During the training process, I've noticed that the loss stops decreasing after around 4000 steps. It stabilizes between 0.2 and 0.3 and makes no further progress.

Here is a screenshot of my training loss curve for your reference:

<img width="769" height="407" alt="Image" src="https://github.com/user-attachments/assets/269cda95-734c-4653-a6de-a0720fe288ca" />

# 2. Evaluation Metrics Do Not Match Paper's Results

Following the training, I used several checkpoints to run the evaluation script. However, the MMD、COV and JSD scores I obtained are significantly different from the results reported in the paper. For instance, my best CD score is 

<img width="1169" height="414" alt="Image" src="https://github.com/user-attachments/assets/3d96ad63-7f2e-474c-8341-6c1445f73936" />

, whereas the paper reports 

<img width="1268" height="282" alt="Image" src="https://github.com/user-attachments/assets/61d65791-1b6d-4791-ace7-f297b7346aa7" />

# What I Have Already Checked:

To ensure this wasn't a simple setup error, I have carefully gone through the following checks:

- Environment: I have created a clean environment using conda and installed all dependencies with the exact versions specified in environments.yaml.
- Data: The dataset was downloaded and preprocessed according to the instructions in the README.md.
- Configuration: I am using the default configuration file provided for this experiment and have double-checked that all hyperparameters match the paper's settings.
- Execution Command: I am using the exact training and evaluation python script provided in your documentation.

# My Question:

Given the details above, could you please suggest any other potential areas I might have overlooked? For instance, could this be related to:
- A subtle detail in the data preprocessing?
- Specific hardware dependencies (I am using an RTX8000 GPU)?
- Any advice or pointers you could offer would be immensely helpful.

## My Environment Details:

- OS: Ubuntu 22.04
- Python Version: 3.9
- PyTorch Version: 2.2.1+cu121
- CUDA Version: 12.3
- GPU: RTX8000

Thank you for your time and consideration. I look forward to hearing from you.

**Comment by ShizhaoSun** [comment]
Hi vegetableno1,

Thanks for your interest in reproducing our work! I’m sorry I don’t have a concrete idea from the experimental details you shared. If I ran into a similar issue, I’d probably start by checking whether the predicted sequence is valid and then take a closer look at the qualitative results (like the rendered CAD models) to see if that gives me more clues.

Thanks

**Comment by Zhanwei-Z** [comment]
Hi vegetableno1,

Thank you for your interest in our work. The issue you raised may be attributed to two main factors:

(1) The original test set contains a number of invalid CAD sequences. If such sequences are included in evaluation, they can significantly degrade performance metrics such as MMD, COV, and JSD. To ensure fair and reliable comparisons, we applied a filtering step to the test set before evaluating both the baselines and our FlexCAD.
(2) Some of the predicted sequences, particularly at the sketch and extrusion levels, may themselves be invalid, a problem that is more significant in the baseline methods (i.e., SkexGen and Hnc-cad). To address this, we perform a post-processing filtering step on all predicted sequences (for both baselines and our FlexCAD) to ensure their validity and renderability before evaluation.

These filtering steps are crucial for obtaining meaningful and fair quantitative comparisons across all methods.
Thanks!
